.pre-commit-config.yaml

```diff
@@ -63,7 +63,7 @@ repos:
             '--extensions=c,h',
             '--headers=h',
             --recursive,
-            '--filter=-readability/casting,-runtime/int,-build/include_subdir'
+            '--filter=-readability/casting,-runtime/int,-build/include_subdir,-readability/fn_size'
```

Comment:
I think this is a good check to keep in place - otherwise these functions get unwieldy

Reply:
Unfortunately the function is now 522 lines long, whereas the limit for this check is 500

Is it OK to turn it off now, or would you prefer a precursor PR to split up this function?

Reply:
Hmm not a great solution here. I think OK for now but something we should take care of in a follow up.

Ideally you could change numpy upstream to split the function (maybe split into a date / time parsing functions?). That way we wouldn't diverge too far from them when we bring that downstream

Reply:
OK I'll see if I can upstream something, thanks!